resources/asciidoctor/lib/definition_hover/extension.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'asciidoctor/extensions'
+
+##
+# Extension for adding an inline definition to key terms in the docs
+#
+# Usage
+#
+#   Foo definition:["word", "definition"]
```

Comment:
I wouldn't use "definition" in the example, since it might confuse people as to what is expected to be copied and what people can change.
```suggestion
#   This is a paragraph about definition:[Elasticsearch, "a distributed, JSON-based search and analytics engine"].
```